src/project_schema/build/index.ts

```diff
@@ -1,8 +1,12 @@
-import { FolderGenerator, TemplateGenerator } from 'src/generators';
+import { FolderGenerator, TemplateGenerator, GenerationContext } from 'src/generators';
+import { srcExt, TemplateOptions } from 'src/template_options';
 
 
 export default new FolderGenerator('build', [
-  new TemplateGenerator(require('./webpack.base.conf.js.ejs'), 'webpack.base.conf.js'),
+  new TemplateGenerator(require('./webpack.base.conf.js.ejs'),
+  'webpack.base.conf.js',
+  (ctx: GenerationContext<TemplateOptions>) => { ctx['extension'] = srcExt('{ext}')(ctx) }

```

Comment:
- indent
```suggestion
  ctx => { ctx['extension'] = srcExt('{ext}')(ctx) }
```

Reply:
it throw error:
```
[ts]
Argument of type 'GenerationContext<{}>' is not assignable to parameter of type 'GenerationContext<TemplateOptions>'.
  Type '{}' is not assignable to type 'TemplateOptions'.
    Property 'id' is missing in type '{}'.
```

Reply:
you should write `TemplateGenerator<TemplateOptions>` then